### DISCUSSION TITLE
JobDataMap put not work

### DISCUSSION THREAD

**Discussion by ZLVincent** [quartznet/quartznet#1823] [Q&A] [answered]
**Describe the bug**

Update the JobDataMap, but it is not work.

**Version used**

3.4.0

**To Reproduce**

var jobDetail = await _scheduler.GetJobDetail(key);
jobDetail.JobDataMap.Put("test", 111);

**Expected behavior**

I has update the JobDataMap for a paused job with the above code.  When I resumed the job, JobDataMap does not have test in the IJobExecutionContext obtained in the Execute method.

**Comment by drieseng** [comment]
@ZLVincent, what job store are you using?

**Reply by ZLVincent** [reply]
> @ZLVincent, what job store are you using?

Quartz.Impl.AdoJobStore.JobStoreTX, Quartz

**Comment by ZLVincent** [comment]
Does anyone else have that requirement? Modify JobDataMap in the paused state

**Comment by lahma** [comment, marked as answer]
You cannot modify the `JobDataMap` and expect it to change in database. `JobDataMap` changes are only automatically updated when done insde `IJob.Execute`.

If you want to update the map for an existing job as a maintenance operation, you need to call `await IScheduler.AddJob(myUpdatedJob, replace: true);`

**Reply by ZLVincent** [reply]
IJobDetail jobDetail = await _scheduler.GetJobDetail(key);
jobDetail.JobDataMap.Put(dataKey, value);
System.Collections.Generic.IReadOnlyCollection<ITrigger> jobTriggers = await _scheduler.GetTriggersOfJob(key);
await _scheduler.ScheduleJob(jobDetail, jobTriggers, replace: true);
